### PR TITLE
docs(s3-deployment): update development section of readme

### PR DESCRIPTION
**Pull request by neo01124**
…ent README

### Reason for this change

The links in the development section of the aws-s3-deployment README are pointing to incorrect locations (files have been moved from those locations), fixed the links.

### Description of changes

There are no code changes, only documentation update

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by scanlonp — DISMISSED** [review]
Nice find @neo01124! I think we can make it even clearer by calling out that the code in now in the custom-resource-handlers package. I think either in the link display text, or just "the code is now in the `@aws-cdk/custom-resource-handlers` package under ...".

**Comment by neo01124** [comment]
sounds good, will add that

File: packages/aws-cdk-lib/aws-s3-deployment/README.md [hunk 1]
```diff
@@ -489,8 +489,8 @@ new cdk.CfnOutput(this, 'ObjectKey', {
 ## Development
 
 The custom resource is implemented in Python 3.9 in order to be able to leverage
-the AWS CLI for "aws s3 sync". The code is under [`lib/lambda`](https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk-lib/aws-s3-deployment/lib/lambda) and
-unit tests are under [`test/lambda`](https://github.com/aws/aws-cdk/tree/main/packages/aws-cdk-lib/aws-s3-deployment/test/lambda).
+the AWS CLI for "aws s3 sync". The code is under [`/lib/aws-s3-deployment/bucket-deployment-handler`](https://github.com/aws/aws-cdk/tree/main/packages/@aws-cdk/custom-resource-handlers/lib/aws-s3-deployment/bucket-deployment-handler/) and
```
Comment (scanlonp) [inline packages/aws-cdk-lib/aws-s3-deployment/README.md:492]:
```suggestion
the AWS CLI for "aws s3 sync".
The code is now in the `@aws-cdk/custom-resource-handlers` package under [`lib/aws-s3-deployment/bucket-deployment-handler`](https://github.com/aws/aws-cdk/tree/main/packages/@aws-cdk/custom-resource-handlers/lib/aws-s3-deployment/bucket-deployment-handler/) and
```

File: packages/aws-cdk-lib/aws-s3-deployment/README.md [hunk 2]
```diff
@@ -489,8 +489,8 @@ new cdk.CfnOutput(this, 'ObjectKey', {
...
+unit tests are under [`/lib/aws-s3-deployment/bucket-deployment-handler/test`](https://github.com/aws/aws-cdk/tree/main/packages/@aws-cdk/custom-resource-handlers/test/aws-s3-deployment/bucket-deployment-handler/).
```
Comment (scanlonp) [inline packages/aws-cdk-lib/aws-s3-deployment/README.md:493]:
```suggestion
unit tests are under [`test/aws-s3-deployment/bucket-deployment-handler`](https://github.com/aws/aws-cdk/tree/main/packages/@aws-cdk/custom-resource-handlers/test/aws-s3-deployment/bucket-deployment-handler/).
```

**Review by scanlonp — COMMENTED** [review]
Finishing this up.

**Review by scanlonp — APPROVED** [review]
Looks good to me! Thanks!

**Review by scanlonp — APPROVED** [review]

**Pull request merged**